src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/BookBinder.java

```diff
@@ -0,0 +1,143 @@
+
+                    EnchantmentStorageMeta enchantMeta = (EnchantmentStorageMeta) book.getItemMeta();
+                    for (Map.Entry<Enchantment, Integer> entry : enchantments.entrySet()) {
+                        enchantMeta.addStoredEnchant(entry.getKey(), entry.getValue(), bypassVanillaMaxLevel.getValue());
+                    }
+
+                    book.setItemMeta(enchantMeta);
+                    
+                    MachineRecipe recipe = new MachineRecipe(25 * (enchantments.size() / this.getSpeed()), new ItemStack[] {target, item}, new ItemStack[] {book});
+
+                    if (!InvUtils.fitAll(menu.toInventory(), recipe.getOutput(), getOutputSlots())) {
+                        return null;
+                    }
+
+                    for (int inputSlot : getInputSlots()) {
+                        menu.consumeItem(inputSlot);
+                    }
+
+                    return recipe;
+                }
+
+                return null;
+            }
+
+        }
+
+        return null;
+    }
+
+    private boolean isCompatible(@Nullable ItemStack item) {
+        return item != null && item.getType() == Material.ENCHANTED_BOOK;
+    }
+
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.IRON_CHESTPLATE);
+    }
+
+    @Override
+    public String getMachineIdentifier() {
+        return "BOOK_BINDER";
+    }
+
+    @Nonnull
+    @ParametersAreNonnullByDefault
+    private Map<Enchantment, Integer> combineEnchantments(Map<Enchantment, Integer> ech1, Map<Enchantment, Integer> ech2) {
+        Map<Enchantment, Integer> enchantments = new HashMap<>();
+        boolean conflicts = false;
+        
+        enchantments.putAll(ech1);
+        for (Map.Entry<Enchantment, Integer> entry : ech2.entrySet()) {
+            for (Map.Entry<Enchantment, Integer> conflictsWith : enchantments.entrySet()) {
+                if (entry.getKey().conflictsWith(conflictsWith.getKey())) {
+                  conflicts = true;
+                }
+            }
+            
+            if (!conflicts) {
+                enchantments.merge(entry.getKey(), entry.getValue(), (a, b) -> {
+                    if (a == b) {
+                        if (hasCustomMaxLevel.getValue()) {
+                            return a + 1 > customMaxLevel.getValue() ? customMaxLevel.getValue() : a + 1;
+                        } else {
+                            return a + 1;
+                        }
+                        
+                    } else {
+                        int highestLevel = Math.max(a, b);
+
+                        if (hasCustomMaxLevel.getValue()) {
+                            return highestLevel > customMaxLevel.getValue() ? customMaxLevel.getValue() : highestLevel;
+                        } else {
+                            return highestLevel;
+                        }
+                    
+                    }
+                });                
+            }   
```

Comment:
This could probably be optimized a lot.
But I think we were holding this back for long enough and can always improve it later.